projects/core/src/cms/facade/page-meta.service.ts

```diff
@@ -43,6 +56,10 @@ export class PageMetaService {
   getMeta(): Observable<PageMeta | null> {
     return this.cms.getCurrentPage().pipe(
       filter(Boolean),
+      tap((page: Page) =>
+        // add dynamic attributes to HTML body element
+        this.dynamicAttributeService.addAttributesToHtmlBody(page)
+      ),
```

Comment:
`addPageContract` function in SmartEditService was deprecated. So, `addAttributesToHtmlBody` must be called somewhere. I add it in `getMeta()` function. I am not sure whether here is a good place to call the function to add attributes to `<body>`